Sources/ComposableArchitecture/Reducer/Reducers/OnChange.swift

```diff
@@ -1,4 +1,58 @@
 extension Reducer {
+  /// Adds a reducer to run when this reducer changes the given value in state.
+  ///
+  /// Use this operator to trigger additional logic when a value changes, like when a
+  /// ``BindingReducer`` makes a deeper change to a struct held in ``BindingState``.
+  ///
+  /// ```swift
+  /// struct Settings: Reducer {
+  ///   struct State {
+  ///     @BindingState var userSettings: UserSettings
+  ///     // ...
+  ///   }
+  ///
+  ///   enum Action: BindableAction {
+  ///     case binding(BindingAction<State>)
+  ///     // ...
+  ///   }
+  ///
+  ///   var body: some Reducer<State, Action> {
+  ///     BindingReducer()
+  ///       .onChange(of: \.userSettings.isHapticFeedbackEnabled, didChange: ==) { oldValue, newValue in
+  ///         Reduce { state, action in
+  ///           .run { send in
+  ///             // Persist new value...
+  ///           }
+  ///         }
+  ///       }
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// When the value changes, the new version of the closure will be called, so any captured values
+  /// will have their values from the time that the observed value has its new value. The system
+  /// passes the old and new observed values into the closure.
+  ///
+  /// > Note: Take care when applying `onChange(of:)` to a reducer, as it adds an equatable check
+  /// > for every action fed into it. Prefer applying it to leaf nodes, like ``BindingReducer``,
+  /// > against values that are quick to equate.
+  ///
+  /// - Parameters:
+  ///   - toValue: A closure that returns a value from the given state.
+  ///   - didChange: A closue that indicates that the value has changed.
+  ///   - reducer: A reducer builder closure to run when the value changes.
+  ///   - oldValue: The old value that failed the comparison check.
+  ///   - newValue: The new value that failed the comparison check.
+  /// - Returns: A reducer that performs
+  @inlinable
+  public func onChange<V, R: Reducer>(
+    of toValue: @escaping (State) -> V,
+    didChange: @escaping (V, V) -> Bool,
```

Comment:
The `didChange` naming makes it sounds like an action closure to execute when something does change. I don't know if there is a precedent for naming this kind of thing in SwiftUI, but at least Combine as `removeDuplicates`. Would you mind updating to that until we figure out something better?

Reply:
What's your take on `isEqual`?

Reply:
Stephen and I will discuss, but in some sense duplicates are being removed. Duplicate invocations of the `reducer` closure are "removed". In general we prefer to not create new naming conventions if it can be helped, and I'm not sure there is a precedent for `isEqual` in Swift or Apple's frameworks.

Reply:
`isEqual` is definitely oldschool Obj-C iOS 2.0+ naming. 😄 

Renamed to `removeDuplicates predicate: @escaping (Value, Value) -> Bool`. 

References:

```swift
func removeDuplicates(by predicate: @escaping (Self.Output, Self.Output) -> Bool
```
[RemoveDuplicates](https://developer.apple.com/documentation/combine/publishers/removeduplicates)<Self>

```swift
func isEqual(_ object: Any?) -> Bool
```
[isEqual](https://developer.apple.com/documentation/objectivec/nsobjectprotocol/1418795-isequal)